File: elem/multiplication_table.js
const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout  
});


const table = () => {
    let horiz = [' '];

    rl.question('Enter a number between 1 and 12: ', function(n) {
        for (let x = 1; x <= n; x++) {
            horiz.push(x);
        }
        console.log(horiz.join('  '));
        for (let x = 1; x <= n; x++) {
            let vert = [];
            for (let y = 0; y <= n; y++) {
                if (y == 0) {
                    vert.push(x);
                } else {
                    vert.push(x * y);
                }
            }
            console.log(vert.join('  '));
        }
        rl.close();
    })
}

rl.on('close', function() {
    process.exit(0);
});

table();

File: elem/multiplication_table.py


def table():
    horizontal = [' ']
    print('Enter a number between 1 and 12: ')
    num = int(input())
    rng = range(1, num + 1)

    for n in rng:
        horizontal.append(n)

    result = "  ".join(str(v) for v in horizontal)
    print(result)

    x = 1
    while x <= num:
        y = 0
        vert = []
        while y <= num:
            if y == 0:
                vert.append(x)
            else:
                vert.append(x * y)
            y += 1
        temp = '  '.join(str(v) for v in vert)
        print(temp)

        x += 1


table()